pkg/datastore/test/basic.go

```diff
@@ -0,0 +1,24 @@
+package test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func UseAfterCloseTest(t *testing.T, tester DatastoreTester) {
+	require := require.New(t)
+
+	// Create the datastore.
+	ds, err := tester.New(0, veryLargeGCInterval, veryLargeGCWindow, 1)
+	require.NoError(err)
+
+	// Immediately close it.
+	err = ds.Close()
+	require.NoError(err)
+
+	// Attempt to use and ensure an error is returned.
+	_, err = ds.HeadRevision(context.Background())
```

Comment:
doesn't this only show that `HeadRevision` can't be used after close?

Reply:
Yes; we could add an exhaustive list but since most operations start with revision checking, I expect any other reader operations will fail too

Reply:
@josephschorr wouldnt `OptimizedRevision` arguably be used more often than `HeadRevision`?

Reply:
As I said, I can add more methods, but ultimately the only true way to be certain would be to test them all. I just wanted to verify that the standard path would result in an error